visualization/app/codeCharta/codeCharta.service.spec.ts

```diff
@@ -154,39 +158,85 @@ describe("codeChartaService", () => {
 			expect(CodeChartaService.ROOT_PATH).toEqual(`/${expected.map.name}`)
 		})
 
-		it("should not replace the last files when loading new files", () => {
-			codeChartaService.loadFiles([
-				{
-					fileName: "FirstFile",
-					content: validFileContent,
-					fileSize: 42
-				}
+		it("should replace files with equal file name and checksum when loading new files", async () => {
```

Comment:
According the bug description of the issue, we should add a regression test to handle that case (uploading broken file, fix it manually, upload again).